src/openssl_wrap.cpp

```diff
@@ -1557,5 +1557,54 @@ std::vector<uint8_t> _EVP_derive_key(const EVP_PKEY *peerkey, const EVP_PKEY *ke
     return result;
 }
 
+SSL_ENGINE_Ptr _ENGINE_by_id(const std::string &engine_id)
+{
+    return SSL_ENGINE_Ptr{
+            OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_ENGINE_by_id, engine_id.c_str())};
+}
+
+void _ENGINE_init(ENGINE *e) { OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_ENGINE_init, e); }
+
+void _ENGINE_ctrl_cmd(ENGINE *e, const std::string &cmd_name, void *opaque_value)
+{
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_ENGINE_ctrl_cmd,
+                                  e,
+                                  /* yes, it's ugly but OpenSSL expect a pointer to unsigned char
+                                   * that is pointing to non-const */
+                                  const_cast<char *>(cmd_name.c_str()),
+                                  0,
+                                  opaque_value,
+                                  nullptr,
+                                  1);
```

Comment:
Can you please add a comment why set to "1"?

From the documentation (man SSL_NEGINE_ctrl_cmd) it is not obvious what the  value is.

Reply:
If cmd_optional is set to 1, it means that the Engine does not need to support the passed (it is optional). I think the value should actually be set to 0 for better guarantees. Fixing...

Reply:
Done.